### PR TITLE
Add status group to User model

### PR DESCRIPTION
**Pull request by MrSerth**
In the future, access to some information on CodeHarbor might be restricted due to the status group. For example, only teachers could get access to model solutions.

Further, we need this extension for #1326 and the NBP project.

### CODE REVIEW

File: db/migrate/20240531160738_add_status_group_to_user.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddStatusGroupToUser < ActiveRecord::Migration[7.1]
+  def change
+    add_column :users, :status_group, :integer, null: false, limit: 2, default: 0, comment: 'Used as enum in Rails'
```
Comment (Mathis-Z) [inline db/migrate/20240531160738_add_status_group_to_user.rb:5]: This could also be "limit: 1", right?

**Review by Mathis-Z — COMMENTED** [review]
I am not sure if we should allow users to specify their status group. We would have to remove this once "Mein Bildungsraum" adds support for a verified status group. Maybe we should just leave it at 'unkown' to avoid getting 'dirty' data.

File: db/migrate/20240531160738_add_status_group_to_user.rb (hunk 1 above)
Comment (MrSerth) [reply db/migrate/20240531160738_add_status_group_to_user.rb]: Yes, indeed. I changed it here (and in the `schema.rb`).

**Comment by MrSerth** [comment]
> I am not sure if we should allow users to specify their status group. We would have to remove this once "Mein Bildungsraum" adds support for a verified status group. Maybe we should just leave it at 'unkown' to avoid getting 'dirty' data.

Nothing easier than that 😁. I agree that it might be useful to avoid potentially incorrect data entered by users manually. Then, getting verified as a teacher is our first feature only available through the integration with "Mein Bildungsraum" 👍.

**Comment by Mathis-Z** [comment]
Okay, I think we can merge the PR :+1:

**Comment by MrSerth** [comment]
@Mathis-Z I saw that you restored the branch -- with a specific reason in mind or my mistake?

**Comment by Mathis-Z** [comment]
I needed it for something but I already forgot for what. I will delete it again.

**Pull request merged**